pkg/controller/operator/master/reconciler.go

```diff
@@ -483,10 +484,14 @@ func (r *Reconciler) reconcileAddonConfigs(ctx context.Context, config *kubermat
 func (r *Reconciler) reconcileApplicationDefinitions(ctx context.Context, config *kubermaticv1.KubermaticConfiguration, logger *zap.SugaredLogger) error {
 	logger.Debug("Reconciling ApplicationDefinitions")
 
-	reconcilers := []kkpreconciling.NamedApplicationDefinitionReconcilerFactory{
-		cilium.ApplicationDefinitionReconciler(config),
+	sysAppDefReconcilers, err := applicationdefinitions.SystemApplicationDefinitionReconcilerFactories(logger, config)
+	if err != nil {
+		return fmt.Errorf("failed to get system application definition reconciler factories: %w", err)
 	}
-	if err := kkpreconciling.ReconcileApplicationDefinitions(ctx, reconcilers, "", r.Client); err != nil {
+
+	sysAppDefReconcilers = append(sysAppDefReconcilers, cilium.ApplicationDefinitionReconciler(config))
+
+	if err := kkpreconciling.ReconcileApplicationDefinitions(ctx, sysAppDefReconcilers, "", r.Client); err != nil {
 		return fmt.Errorf("failed to reconcile ApplicationDefinitions: %w", err)
 	}
 
```

Comment:
Small request:

Please bring back the functionality to disable system applications. Since we are going installer-less for this, you can introduce this in the KubermaticConfiguration, could be something like this:
```go
type SystemApplicationOptions struct {
	// Disable is used to disable the installation of system application definitions in the master cluster.
	Disable bool `json:"disable,omitempty"`
	// Applications is a list of system application definitions that should be installed in the master cluster.
	Applications []string `json:"applications,omitempty"`
}
```

Reply:
The idea of `SystemApplicationOptions.Applications` is similar to https://github.com/kubermatic/kubermatic/issues/14232, so that users can limit the controller to only install certain system applications.

Reply:
System application shouldn't be disabled, since we explicitly say they are managed by KKP, thus users shouldn't be able to disable them at least at the operator level. Disable the cluster autoscaler by users, would lead into a bug in the UI which is, the checkbox in the machine deployment step. So if someone for whatever reason disables this, it would lead us to the exact same problem again. Unless this is fix in the UI I would say we shouldn't enable/disable system applications. Maybe in further release we can come up with a better approach but for now please do not give the users the possibility to disable system apps.